### PR TITLE
DEVPROD-9423: create job to sync project vars to Parameter Store

### PR DESCRIPTION
**Pull request by Kimchelly**
DEVPROD-11187

### Description
Project vars can use Parameter Store now, but they need to be migrated into Parameter Store first. This temporary job aids the migration by automatically syncing the project vars into Parameter Store whenever it detects that a project has Parameter Store enabled but its vars are not in sync.

### Testing
Tested it on a branch project and repo project in staging, and it synced their vars as expected.

### Documentation
N/A

### CODE REVIEW

File: model/repo_ref.go [hunk 1]
```diff
@@ -88,6 +89,20 @@ func FindRepoRefByOwnerAndRepo(owner, repoName string) (*RepoRef, error) {
 	}))
 }
 
+// FindRepoRefsToSync finds all repo refs that have Parameter Sore enabled but
+// don't have their project variables synced to Parameter Store yet.
+func FindRepoRefsToSync(ctx context.Context) ([]RepoRef, error) {
```
Comment (hadjri) [inline model/repo_ref.go:94]: Does DEVPROD-11882 apply to this function?

File: model/project_ref.go [hunk 2]
```diff
@@ -3685,3 +3686,25 @@ func (p *ProjectRef) setParameterStoreVarsSynced(isSynced bool, isRepoRef bool)
...
+	projectRefParameterStoreEnabledKey: true,
+	"$or": []bson.M{
+		{projectRefParameterStoreVarsSyncedKey: false},
+		{projectRefParameterStoreVarsSyncedKey: bson.M{"$exists": false}},
+	},
+}
+
+// FindProjectRefsToSync finds all project refs that have Parameter Sore enabled
+// but don't have their project variables synced to Parameter Store yet.
+func FindProjectRefsToSync(ctx context.Context) ([]ProjectRef, error) {
```
Comment (hadjri) [inline model/project_ref.go:3700]: Same q here

File: model/repo_ref.go (hunk 1 above)
Comment (Kimchelly) [reply model/repo_ref.go:94]: Yes, this is temporary logic.

File: model/project_ref.go (hunk 2 above)
Comment (Kimchelly) [reply model/project_ref.go:3700]: Same as above.

**Review by hadjri — APPROVED** [review]

**Comment by hadjri** [comment]
Can we slap a TODO on those function comments so we don't forget to remove them?

**Pull request merged**